Fix ctypes(3) usage, toupper(3) expects an int and passing a signed char may
lead to UB.

--- src/logging.c~	2026-05-06 20:34:17.000000000 +0200
+++ src/logging.c	2026-06-04 19:38:35.196522630 +0200
@@ -895,7 +895,7 @@
     return RCUTILS_RET_BAD_ALLOC;
   }
   for (int i = 0; severity_string_upper[i]; ++i) {
-    severity_string_upper[i] = (char)toupper(severity_string_upper[i]);
+    severity_string_upper[i] = (char)toupper((unsigned char)severity_string_upper[i]);
   }
 
   // Determine the severity value matching the severity name.
